recipes/libsndfile/all/conanfile.py

```diff
@@ -43,55 +42,36 @@ def requirements(self):
-        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
+        tools.get("https://github.com/erikd/libsndfile/archive/v{0}.tar.gz".format(self.version))
         extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        os.rename(extracted_dir, self._source_subfolder())
 
-    def _configure_autotools(self):
-        if not self._autotools:
-            self._autotools = AutoToolsBuildEnvironment(self)
-            args = [
-                "--without-caps",
-                "--enable-alsa" if self.options.with_alsa else "--disable-alsa",
-                "--enable-sqlite" if self.options.with_sqlite else "--disable-sqlite",
-                "--enable-external-libs" if self.options.with_external_libs else "--disable-external-libs",
-            ]
-            if self.options.shared:
-                args.extend(['--enable-shared=yes', '--enable-static=no'])
-            else:
-                args.extend(['--enable-shared=no', '--enable-static=yes'])
-            self._autotools.configure(args=args, configure_dir=self._source_subfolder)
-        return self._autotools
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+        self._cmake.definitions["BUILD_REGTEST"] = self.options.with_sqlite
+        self._cmake.definitions["ENABLE_EXTERNAL_LIBS"] = self.options.with_external_libs
+        self._cmake.configure(source_folder=self._source_subfolder())
+        return self._cmake
 
     def build(self):
-        if self.options.with_external_libs:
-            shutil.copyfile('vorbis.pc', 'vorbisenc.pc')
-        with tools.environment_append(RunEnvironment(self).vars):
-            autotools = self._configure_autotools()
-            autotools.make()
+        cmake = self._configure_cmake()
+        cmake.build()
 
     def package(self):
-        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
-        autotools = self._configure_autotools()
-        autotools.install()
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        for f in glob.glob(os.path.join(self.package_folder, "lib", "*.la")):
-            os.remove(f)
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder())
+        cmake = self._configure_cmake()
+        cmake.install()
 
     def package_info(self):
-        self.cpp_info.names['pkg_config'] = 'sndfile'
-        self.cpp_info.libs = tools.collect_libs(self)
-        if self.settings.os == "Linux":
-            self.cpp_info.system_libs = ["m", "dl", "pthread", "rt"]
-
+        self.cpp_info.libs = ["sndfile"]
```

Comment:
It's super that you list the libraries explicitly, instead of collecting them, using `tools.collect_libs`.
But you will also need to set other properties, as is done in the autotools recipe.
(When copying, please change the single quotes to double quotes to remain consistent)